Scripts/Python/ki/xKIChat.py

```diff
@@ -439,9 +440,20 @@ def AddChatLine(self, player, message, cFlags, forceKI=True):
 
         # Is it an object to represent the flags?
         if isinstance(cFlags, ChatFlags):
+            PtDebugPrint("Chat Flags are {}".format(cFlags))
+            # Is it subtitles for current audio?
+            if cFlags.subtitle:
+                headerColor = kColors.AudioSubtitleHeader
+                player = None
+
+                PtDebugPrint("xKIChat.AddChatLine(): Adding subtitle {}.".format(message))
+                messageMatches = SUBTITLE_REGEX.match(message)
+                if messageMatches is not None:
+                    pretext = messageMatches.group("speaker")
+                    message = messageMatches.group("subtitle")
```

Comment:
That seems reasonable, but I'll need to think a bit on what the right way to do that is. AddChatLine takes 3 arguments: player, message, and chatFlags. Maybe I can hack in the ability to use "player" for the speaker name string (if provided) and just know to treat it as a string rather than a ptPlayer when the chatFlags indicate the message is a subtitle? It's a bit non-obvious, I think. If you want, I can try it and push the regex matching up to the plSrtFileReader-level to split out the subtitle lines into speaker and text.